src/main/kotlin/com/karumi/maxibonkata/KarumiHQs.kt

```diff
@@ -0,0 +1,33 @@
+package com.karumi.maxibonkata
+
+class KarumiHQs(val chat: Chat = ConsoleChat()) {
+    var maxibonsLeft: Int = 10
+
+    fun openFridge(developer: Developer) {
+        grabMaxibons(developer)
+        if (shouldBuyMoreMaxibons()) {
+            notifyWeShouldBuyMaxibon(developer)
+            buyMaxibons()
+        }
+    }
+
+    fun openFridge(developers: List<Developer>) {
+        developers.forEach { developer ->
+            openFridge(developer)
+        }
+    }
+
+    private fun grabMaxibons(developer: Developer) {
+        maxibonsLeft -= developer.maxibonsToGrab
+        maxibonsLeft = Math.max(maxibonsLeft, 0)
+    }
+
+    private fun shouldBuyMoreMaxibons(): Boolean = maxibonsLeft <= 2
+
+    private fun notifyWeShouldBuyMaxibon(developer: Developer) =
+            chat.sendMessage("Hi guys, I'm " + developer.name + ". We need more maxibons!")
```

Comment:
https://kotlinlang.org/docs/reference/idioms.html#string-interpolation 😅

Reply:
Yeap, you are totally right. I've updated the PR using string interpolation 😃